hydroflow/examples/deadlock_detector/helpers.rs

```diff
@@ -0,0 +1,37 @@
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use std::net::SocketAddr;
+use tokio_util::codec::LinesCodecError;
+
+pub fn serialize_msg<T>(msg: T) -> String
+where
+    T: Serialize + for<'a> Deserialize<'a> + Clone,
```

Comment:
(Can use [`serde::de::DeserializeOwned`](https://docs.rs/serde/latest/serde/de/trait.DeserializeOwned.html#) instead of `for<'a> Deserialize<'a>`)
Is `Clone` needed?

Reply:
noted and Clone removed